Upgrade to ipfs@0.41 (#381)
ProtoSchool/protoschool.github.io: src/tutorials/0005-regular-files-api/03.md

Let's take a look at how to add a file your IPFS node. We'll do this by executing the `add` method:

```javascript
ipfs.add(data, [options])
```

So if we had the `File` object for an adorable photo of a kitten in our browser, accessible to us via a variable `catPic`, and we wanted to add it to our IPFS node, we could pass it into the `add` method as `data` like so:

```javascript
ipfs.add(catPic)
```

Note that the `add` method can accept either a single `File` or an array, just in case you have more than one adorable animal photo to add to the node:

```javascript
ipfs.add([catPic, dogPic, giraffePic])
```

> Because the `add` method returns an [`Async Iterable`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of), you'll need to call the iterable multiple times until you have all the results.

Reviewer: ```suggestion
Because the `add` method returns an [`Async Iterable`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of), you'll need to collect all the results by iterating over it until it is done.
```

For this, we can use a `for await...of` loop:

```javascript
const result = []

for await (const resultPart of ipfs.add([catPic, dogPic, giraffePic])) {
    result.push(resultPart)
}
```

To make things easier, we can use the `it-all` package that does this automatically:

```javascript
const all = require('it-all')

const result = await all(ipfs.add([catPic, dogPic, giraffePic]))
```

The value of the variable `result` is an array of objects, one for each file added to IPFS, in the following format:

```javascript
{
    path: string,
    cid: CID,
    size: number,
    mode: number
}
```

> The value of the `cid` is a CID object (Content Identifier), a unique address generated from the content of the node. (For a more in-depth look at how CIDs are generated and why they're important, check out our [Decentralized data structures](https://proto.school/#/data-structures) tutorial.) In a future lesson, we will learn how to use this value to retrieve the contents of a file.

Reviewer: ```suggestion
The value of the `cid` is a CID object (Content Identifier), a unique address generated from the contents of each file. (For a more in-depth look at how CIDs are generated and why they're important, check out our [Decentralized data structures](https://proto.school/#/data-structures) tutorial.) In a future lesson, we will learn how to use this value to retrieve the contents of a file.
```

Let's take a look at how to add a file your IPFS node. We'll do this by executing the `add` method:

```javascript
ipfs.add(data, [options])
```

So if we had the `File` object for an adorable photo of a kitten in our browser, accessible to us via a variable `catPic`, and we wanted to add it to our IPFS node, we could pass it into the `add` method as `data` like so:

```javascript
ipfs.add(catPic)
```

Note that the `add` method can accept either a single `File` or an array, just in case you have more than one adorable animal photo to add to the node:

```javascript
ipfs.add([catPic, dogPic, giraffePic])
```

> Because the `add` method returns an [`Async Iterable`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of), you can only iterate over the values, one by one. In the case you need to return all the values, you can save each one into an array and return it in the end.

Reviewer 2: ```suggestion
Because the `add` method returns an [`Async Iterable`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of), you can only iterate over the values, one by one. If you need to return all the values, you can save each one into an array and then return the array.
```

> For this, we can use a `for await...of` loop:

Reviewer 2: ```suggestion
To iterate through the Async Iterable, we can use a `for await...of` loop:
```

Author: Same as the previous one
